addon-test-support/-private/async-iterator.js

```diff
@@ -88,20 +91,29 @@ export default class AsyncIterator {
   /**
    * Set a timeout to reject a promise if it doesn't get response within the timeout threshold.
    *
-   * @param {*} reject
+   * @param {*} resolve
    */
-  _setTimeout(reject) {
+  _setTimeout(resolve, reject) {
     clearTimeout(this.timeout);
     this.timer = setTimeout(() => {
       if (!this._waiting) {
         return;
       }
-      let err = new Error(
-        `EmberExam: Promise timed out after ${
+
+      if (this._emberExamExitOnError) {
+        let err = new Error(
+          `EmberExam: Promise timed out after ${
+            this._timeout
+          } s while waiting for response for ${this._request}`
+        );
+        reject(err);
+      } else {
+        // eslint-disable-next-line no-console
+        console.error(`EmberExam: Promise timed out after ${
           this._timeout
-        } s while waiting for response for ${this._request}`
-      );
-      reject(err);
+        } s while waiting for response for ${this._request}. Closing browser to exit gracefully.`);
```

Comment:
should we include "retrying count/maximum" in the message, so it's clear what is up?

Reply:
I don't think we added any retry logic, in that case, do you feel the error message need more to clarify what happened?

Side-note: Although the change in testem to reconnect the socket when a socket disconnect, should help with the problem being addressed by this PR.
With that said, I still feel this PR is still beneficial.